feat: DAYL-96 Update Count Badge a11y docn (#5077)
BrightspaceUI/core: components/count-badge/README.md

| `max-digits` | Number, default: `2` when `type="notification"`, `5` when `type="count"` | Specifies a digit limit, after which numbers are truncated. Defaults to two for `"notification"` type and five for `"count"` type. Must be between 1-5.
| `hide-zero` | Boolean, default: `false` | Hides the count badge when `number` is zero. |
| `text` | String, required | Descriptive text for the badge which will act as an accessible label and tooltip text when tooltips are enabled. |
| `tab-stop` | Boolean, default: `false` |  Adds a tab stop to the badge, which allows screen-reader users to easily tab to the badge. |
| `announce-changes` | Boolean, default: `false` | Announces changes to the badge with an `aria-live` region. If the number property is changed, the text will be read by screenreaders. |
| `has-tooltip` | Boolean, default: `false` | Adds a tooltip on the badge, which will be visible on hover/tab-stop, and read out by screen-readers. |
| `focus-ring` | Boolean, default: `false` | Forces a focus ring around the badge. This property can be used to highlight the badge when the parent element is focused. |
<!-- docs: end hidden content -->

## Accessibility

> - While non-interactable components traditionally shouldn't be focusable, the `tab-stop` property greatly helps screen-reader users find the the count badge
>   - Otherwise, screen-reader users would have to use the arrow keys to get to the badge, which can take considerably longer
>   - If the optional tooltip is used, then `tab-stop` isn't necessary, as the count-badge becomes focusable to render the tooltip
> - The optional tooltip makes use of the [`d2l-toolip`](../../components/tooltip), which follows the [W3C best practices for Tooltips](https://www.w3.org/WAI/ARIA/apg/patterns/tooltip/)
>   - The use of this tooltip can help provide additional context such as providing the exact number, if it were to go above the limit set in `max-digits`

Reviewer: Just trying to make it a bit more clear, though probably needs @geurts input on wording
```suggestion

Notable accessibility features:
- `tab-stop`: 
  - While non-interactable components traditionally shouldn't be focusable, this property greatly helps screen reader users find the the count badge, and removes the need to use the arrow keys to navigate to it.
  - If the optional tooltip is used, it isn't necessary, as the count-badge becomes focusable in order to display the tooltip
- `has-tooltip`: 
  - The optional tooltip makes use of the [`d2l-tooltip`](../../components/tooltip), which follows the [W3C best practices for Tooltips](https://www.w3.org/WAI/ARIA/apg/patterns/tooltip/)
  - This tooltip can help provide additional context, such as by providing the exact count badge number if it were to go above the limit set in `max-digits`
```

Author: While I do like some of the changes, I'm not sure about the formatting, so I'd also likely to hear Jeff's input

Reviewer: Totally fair!

| `max-digits` | Number, default: `2` when `type="notification"`, `5` when `type="count"` | Specifies a digit limit, after which numbers are truncated. Defaults to two for `"notification"` type and five for `"count"` type. Must be between 1-5.
| `hide-zero` | Boolean, default: `false` | Hides the count badge when `number` is zero |
| `text` | String, required | Descriptive text for the badge which will act as an accessible label and tooltip text when tooltips are enabled |
| `tab-stop` | Boolean, default: `false` |  Adds a tab stop to the badge, which allows screen reader and keyboard users to easily tab to the badge |
| `announce-changes` | Boolean, default: `false` | When `true`, changes to the badge will be announced to screen reader users |
| `has-tooltip` | Boolean, default: `false` | Adds a tooltip on the badge, which will be visible on hover/tab-stop, and read out by screen-readers |
| `focus-ring` | Boolean, default: `false` | Forces a focus ring around the badge. This property can be used to highlight the badge when the parent element is focused. |
<!-- docs: end hidden content -->

## Accessibility

Notable accessibility features:
- `tab-stop`:

>   - While non-interactable components traditionally shouldn't be focusable, this property greatly helps screen reader users find the the count badge, and removes the need to use the arrow keys to navigate to it.

Reviewer: ```suggestion
  - While non-interactive components traditionally shouldn't be focusable, this property greatly helps screen reader users find the the count badge and removes the need to use the arrow keys to navigate to it.
```

>   - If the optional tooltip is used, it isn't necessary, as the count-badge becomes focusable in order to display the tooltip

Reviewer: ```suggestion
  - This isn't needed if the optional tooltip is used, as the count-badge becomes focusable in order to display the tooltip.
```

Author: I've made all these changes, with the exception of having the period at the end of the line here, since I remember you mentioning that we shouldn't be including them. Unless there's a special case when it's a longer sentence?

Reviewer: I added it here to be consistent with the line above. I do wonder about the longer sentence piece. Thoughts @geurts ?

Reviewer: Based on https://daylight.d2l.dev/style-elements/writing/#periods it should probably just be left off, so LGTM
